api/src/main/java/org/apache/iceberg/types/CheckCompatibility.java

```diff
@@ -38,7 +38,20 @@
    * @return a list of error details, or an empty list if there are no compatibility problems
    */
   public static List<String> writeCompatibilityErrors(Schema readSchema, Schema writeSchema) {
-    return TypeUtil.visit(readSchema, new CheckCompatibility(writeSchema, true));
+    return TypeUtil.visit(readSchema, new CheckCompatibility(writeSchema, true, false));
+  }
+
+  /**
+   * Returns a list of compatibility errors for writing with the given write schema.
+   *
+   * @param readSchema a read schema
+   * @param writeSchema a write schema
+   * @param disableNullabilityCheck disable the check for nullable on fields
+   * @return a list of error details, or an empty list if there are no compatibility problems
+   */
+  public static List<String> writeCompatibilityErrors(Schema readSchema, Schema writeSchema,
+                                                      Boolean disableNullabilityCheck) {
```

Comment:
I think this should be `checkNullability`. It's usually easier to understand if boolean are phrased as positive instead of negative: "if not check nullability" vs "if not disable nullability check".